src/main/java/io/kontur/disasterninja/service/GeometryTransformer.java

```diff
@@ -56,4 +61,14 @@ public static boolean geometriesAreEqual(org.wololo.geojson.Geometry geometry1,
                                              org.wololo.geojson.Geometry geometry2) {
         return JsonUtil.writeObjectNode(geometry1).equals(JsonUtil.writeObjectNode(geometry2));
     }
+
+    public Geometry buffer(Geometry geometry, double buffer) {
+        if (geometry == null) {
+            return null;
+        }
+        if ("Polygon".equals(geometry.getType()) || "MultiPolygon".equals(geometry.getType()) || "GeometryCollection".equals(geometry.getType())) {
```

Comment:
this feels weird. you are only buffering the output if it's polygonal, which only makes sense if you only ever use it as buffer(0). if that is the case then rename the method into MakeValid or check if JTS provides one nowadays (it was recently added to GEOS)

Reply:
Yep, I use it to fix the issues related to crossing polygons. There is no `makeValid` method already implemented in the library. I renamed the method in DN BE to `makeValid` and removed the buffer argument, because it's always 0 for this purpose.

Reply:
https://www.tabnine.com/code/java/methods/org.geotools.geometry.jts.JTS/makeValid ?